Data race in NameResolve.Listener.onError
NameResolve.Listener.onError can be called concurrently in different threads, so the following code in onError() impl may have data race.